pipelines/tekton/test-mlflow-image-pipeline/test-mlflow-image-pipeline.yaml

```diff
@@ -21,82 +21,24 @@ spec:
     params:
     - name: SCRIPT
       value: |
-        cat <<EOF | oc apply -f -
-        apiVersion: apps/v1
-        kind: Deployment
-        metadata:
-          creationTimestamp: null
-          labels:
-            app: $(params.model-name)-$(params.model-version)
-          name: $(params.model-name)-$(params.model-version)
-          namespace: $(params.target-namespace)
-        spec:
-          replicas: 1
-          selector:
-            matchLabels:
-              app: $(params.model-name)-$(params.model-version)
-          strategy: {}
-          template:
-            metadata:
-              creationTimestamp: null
-              labels:
-                app: $(params.model-name)-$(params.model-version)
-            spec:
-              containers:
-              - image: $(params.model-name):$(params.model-version)
-                name: $(params.model-name)-$(params.model-version)
-                livenessProbe:
-                  failureThreshold: 8
-                  httpGet:
-                    path: /v2/health/live
-                    port: 8080
-                    scheme: HTTP
-                  periodSeconds: 5
-                  successThreshold: 1
-                readinessProbe:
-                  failureThreshold: 8
-                  httpGet:
-                    path: /v2/models/$(params.model-name)/ready
-                    port: 8080
-                    scheme: HTTP
-                  periodSeconds: 5
-                  successThreshold: 1
-                ports:
-                - containerPort: 8080
-                resources: {}
-        status: {}
-        EOF
+        echo "$MODEL_DEPLOY_CM" | sed 's/MODEL_NAME/$(params.model-name)/g; s/MODEL_VERSION/$(params.model-version)/g; s/MODEL_IMAGE/$(params.model-name):$(params.model-version)/g' | oc apply -f -
```

Comment:
Would the ACM code still be able to use the files if they were converted to OpenShift Templates? Wouldn't you be forced to use the `oc` at that point instead of patching with kustomize since you would need to pass in params?

Reply:
Given that we'll probably move to using an ACM subscription to deploy the test/stage environment in future, so that it's done in the same way as "production", I don't think it's worth continuing to debate the merits of templates versus configmaps :)